src/transformStateWithClones.js

```diff
@@ -6,8 +6,43 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let clonedState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': {
+        Object.assign(clonedState, action.extraData);
+
+        break;
+      }
+
+      case 'removeProperties': {
+        for (const key of action.keysToRemove) {
+          delete clonedState[key];
+        }
+
+        break;
+      }
+
+      case 'clear': {
+        clonedState = {};
+        break;
+      }
+
+      default: {
+        throw Error(`Unknown action: ${action.type}`);
+      }
+    }
+
+    const clonedStateCopy = { ...clonedState };
```

Comment:
You can directly push the new state into the result array instead of creating a `clonedStateCopy` variable.